File: runs/2026-06-06_0607-UTC/10-meeple/android/app/src/main/kotlin/com/orbioom/meeple/domain/Format.kt
package com.orbioom.meeple.domain

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale
import kotlin.math.roundToInt

/** Pure formatting helpers. Locale.US keeps numbers stable and dot-decimal across devices. */
object Format {

    private val dayFormat = SimpleDateFormat("d MMM yyyy", Locale.US)
    private val shortFormat = SimpleDateFormat("d MMM", Locale.US)
    private val monthFormat = SimpleDateFormat("MMM yyyy", Locale.US)

    fun date(epoch: Long): String = dayFormat.format(Date(epoch))
    fun shortDate(epoch: Long): String = shortFormat.format(Date(epoch))
    fun month(epoch: Long): String = monthFormat.format(Date(epoch))

    /** Trim trailing zeros: 12.0 -> "12", 12.50 -> "12.5", 12.345 -> "12.35". */
    fun number(value: Double, maxDecimals: Int = 2): String {
        if (value.isNaN() || value.isInfinite()) return "0"
        val rounded = String.format(Locale.US, "%.${maxDecimals}f", value)
        return rounded.trimEnd('0').trimEnd('.').ifEmpty { "0" }
    }

    fun integer(value: Double): String = String.format(Locale.US, "%,d", value.toLong())

    /** A 0..1 ratio rendered as a whole-number percent; safe on NaN. */
    fun percent(ratio: Double): String {
        if (ratio.isNaN() || ratio.isInfinite()) return "—"
        return "${(ratio * 100).roundToInt()}%"
    }

    /** A win-loss record like "12–5" (en dash). */
    fun record(wins: Int, losses: Int): String = "$wins–$losses"

    fun duration(minutes: Int?): String {
        if (minutes == null || minutes <= 0) return "—"
        val h = minutes / 60
        val m = minutes % 60
        return when {
            h <= 0 -> "${m}m"
            m == 0 -> "${h}h"
            else -> "${h}h ${m}m"
        }
    }

    /** "12 plays" / "1 play". */
    fun plays(count: Int): String = "$count play${if (count == 1) "" else "s"}"

    fun players(min: Int, max: Int): String =
        if (min == max) "$min player${if (min == 1) "" else "s"}" else "$min–$max players"

    /** "—" when no score, else trimmed number. */
    fun score(value: Double?): String = if (value == null) "—" else number(value, 1)
}

File: runs/2026-06-09_0608-UTC/06-hush/ios/Hush/Utilities/Format.swift
import Foundation

enum Format {
    /// mm:ss (or h:mm:ss past an hour).
    static func clock(_ seconds: Int) -> String {
        let s = max(0, seconds)
        let h = s / 3600
        let m = (s % 3600) / 60
        let sec = s % 60
        if h > 0 { return String(format: "%d:%02d:%02d", h, m, sec) }
        return String(format: "%d:%02d", m, sec)
    }

    static func minutesLabel(_ minutes: Int) -> String {
        if minutes < 60 { return "\(minutes) min" }
        let h = minutes / 60
        let m = minutes % 60
        return m == 0 ? "\(h) h" : "\(h) h \(m) min"
    }

    static func percent(_ value: Double) -> String {
        "\(Int((value * 100).rounded()))%"
    }
}
